clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -164,8 +164,8 @@
     public static final String RETRIES_CONFIG = "retries";
     private static final String RETRIES_DOC = "Setting a value greater than zero will cause the client to resend any record whose send fails with a potentially transient error."
                                               + " Note that this retry is no different than if the client resent the record upon receiving the "
-                                              + "error. Allowing retries will potentially change the ordering of records because if two records are "
-                                              + "sent to a single partition, and the first fails and is retried but the second succeeds, then the second record "
+                                              + "error. Allowing retries without setting <code>max.in.flight.requests.per.connection</code> to 1 will potentially change the ordering of records because if two batches are "
```

Comment:
Maybe we could use the `MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION` variable instead of hardcoding the value here?